### PR TITLE
Add State Management to palette #21507

### PR DESCRIPTION
**Pull request by alfredo-dotcms**
### Proposed Changes
 we need to add State Management to our Content Reuse Palette

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

### CODE REVIEW

**Review by rjvelazco — DISMISSED** [review]

File: apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-palette/store/dot-palette.store.ts [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+    private readonly totalRecords = this.updater((state: DotPaletteState, data: number) => {
+        return { ...state, totalRecords: data };
+    });
+
+    readonly filter = this.updater((state: DotPaletteState, data: string) => {
+        return { ...state, filter: data };
+    });
+
+    readonly languageId = this.updater((state: DotPaletteState, data: string) => {
```
Comment (hmoreras) [inline apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-palette/store/dot-palette.store.ts:167]: updateLanguage or `setLanguageId`

File: apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-palette/store/dot-palette.store.ts [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+        return { ...state, languageId: data };
+    });
+
+    readonly viewContentlet = this.updater((state: DotPaletteState, data: string) => {
```
Comment (hmoreras) [inline apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-palette/store/dot-palette.store.ts:171]: `set...`

**Review by hmoreras — CHANGES_REQUESTED** [review]

File: apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-palette/dot-palette.component.ts [hunk 3]
```diff
@@ -22,28 +26,60 @@ import { DotPaletteContentletsComponent } from './dot-palette-contentlets/dot-pa
...
      * @memberof DotPaletteContentletsComponent
      */
     switchView(variableName?: string): void {
-        this.contentTypeVariable = variableName ? variableName : '';
-        this.stateContentlet = variableName ? 'contentlet:in' : 'contentlet:out';
+        const viewContentlet = variableName ? 'contentlet:in' : 'contentlet:out';
+
+        this.store.setViewContentlet(viewContentlet);
+        this.store.setFilter('');
+        this.store.loadContentlets(variableName);
```
Comment (fmontes) [inline apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-palette/dot-palette.component.ts:62]:
```suggestion
        this.store.switchView(...);
```

And do all the other things internally? We only call all those here.

File: apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-palette/store/dot-palette.store.ts [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+        this.loading$,
+        (contentlets, contentTypes, filter, totalRecords, viewContentlet, loading) => ({
+            contentlets,
+            contentTypes,
+            filter,
+            totalRecords,
+            viewContentlet,
+            loading
+        })
+    );
```
Comment (fmontes) [inline apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-palette/store/dot-palette.store.ts:144]: Remove all the selectors that are not being used and just use the full state object if you want to skip lang id cool, but I would just send it.

**Review by fmontes — CHANGES_REQUESTED** [review]

**Review by fmontes — APPROVED** [review]

**Review by hmoreras — APPROVED** [review]

**Pull request merged**